### ISSUE TITLE
Do not Frozen BN during training?

### ISSUE THREAD

**Issue by yangJirui** [open-mmlab/mmdetection#437]
hey guys, I noticed that 
`class ResNet(nn.Module):

    def __init__(self,
                 depth,
                 num_stages=4,
                 strides=(1, 2, 2, 2),
                 dilations=(1, 1, 1, 1),
                 out_indices=(0, 1, 2, 3),
                 style='pytorch',
                 frozen_stages=-1,
                 normalize=dict(type='BN', frozen=False),  ## Pay attention here.
                 norm_eval=True, ## Pay attention here.
                 dcn=None,
                 stage_with_dcn=(False, False, False, False),
                 with_cp=False,
                 zero_init_residual=True):
`

And I didn't find any modifications for freezing BN in configs(e.g. Faster_RCNN_FPN , RetianNet).

`type='FastRCNN',
pretrained='modelzoo://resnet50',
backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        style='pytorch')
`

It means BN is used during training? 
Usually, BN is frozen in many other codes, because of small batch_size (e.g. <16). Why mmdetection use BN by default ?
Has anyone tried to freeze BN, and has the performance gained ?

**Comment by hellock** [comment]
BN is in the eval mode during training. You can refer to #346

**Comment by yangJirui** [comment]
> BN is in the eval mode during training. You can refer to #346

Thank you for your answer. 
What does Eval mode mean? 
Does it mean neither updating gradient update in norm layers nor updating moving average statistics ?

**Comment by hellock** [comment]
Running mean and variance are not tracked and batch statistics are used instead. It will call `.eval()` for BN module, you can refer to the PyTorch documentation.

**Comment by yangJirui** [comment]
> Running mean and variance are not tracked and batch statistics are used instead. It will call `.eval()` for BN module, you can refer to the PyTorch documentation.

Thank you  for your patient reply.

**Comment by Le1kk** [comment]
> BN is in the eval mode during training. You can refer to #346

Can you tell me what **frozen_stages** indicates?

**Comment by sharkdeng** [comment]
> hey guys, I noticed that
> `class ResNet(nn.Module):
> 
> ```
> def __init__(self,
>              depth,
>              num_stages=4,
>              strides=(1, 2, 2, 2),
>              dilations=(1, 1, 1, 1),
>              out_indices=(0, 1, 2, 3),
>              style='pytorch',
>              frozen_stages=-1,
>              normalize=dict(type='BN', frozen=False),  ## Pay attention here.
>              norm_eval=True, ## Pay attention here.
>              dcn=None,
>              stage_with_dcn=(False, False, False, False),
>              with_cp=False,
>              zero_init_residual=True):
> ```
> 
> `
> 
> And I didn't find any modifications for freezing BN in configs(e.g. Faster_RCNN_FPN , RetianNet).
> 
> `type='FastRCNN', pretrained='modelzoo://resnet50', backbone=dict( type='ResNet', depth=50, num_stages=4, out_indices=(0, 1, 2, 3), frozen_stages=1, style='pytorch') `
> 
> It means BN is used during training?
> Usually, BN is frozen in many other codes, because of small batch_size (e.g. <16). Why mmdetection use BN by default ?
> Has anyone tried to freeze BN, and has the performance gained ?

good question, I got one code using ResnetXT which freeze and another tutorial says freezing can boost performance. Got confused now.
